client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { getAllSubscriptions } from 'calypso/state/memberships/subscriptions/selectors';
+import {
+	getUserPurchases,
+	hasLoadedUserPurchasesFromServer,
+	isFetchingUserPurchases,
+} from 'calypso/state/purchases/selectors';
+import getAvailableConciergeSessions from 'calypso/state/selectors/get-available-concierge-sessions';
+import getConciergeNextAppointment, {
+	NextAppointment,
+} from 'calypso/state/selectors/get-concierge-next-appointment';
+import getConciergeUserBlocked from 'calypso/state/selectors/get-concierge-user-blocked';
+import getSites from 'calypso/state/selectors/get-sites';
+import { getSiteId } from 'calypso/state/sites/selectors';
+import { AppState } from 'calypso/types';
+import MembershipSite from '../membership-site';
+import PurchasesSite from '../purchases-site';
+import { purchasesDataFields } from './purchases-data-field';
+import { purchasesDataView } from './purchases-data-view'; // This is the temporary data
+import PurchasesListHeader from './purchases-list-header';
+import './style.scss';
+
+const useDataViewPurchasesList = config.isEnabled( 'purchases/purchase-list-dataview' );
+
+export interface PurchasesListProps {
+	noticeType?: string | undefined;
+}
+
+export interface PurchasesListConnectedProps {
+	hasLoadedUserPurchasesFromServer: boolean;
+	isFetchingUserPurchases: boolean;
+	purchases: Purchase[] | null;
+	subscriptions: MembershipSubscription[];
+	sites: SiteDetails[];
+	nextAppointment: NextAppointment | null;
+	isUserBlocked: boolean;
+	availableSessions: number[];
+	siteId: number | null;
+}
+
+class PurchasesListDataView extends Component<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps & LocalizeProps
+> {
+	isDataLoading() {
+		if ( this.props.isFetchingUserPurchases && ! this.props.hasLoadedUserPurchasesFromServer ) {
+			return true;
+		}
+
+		return ! this.props.sites.length && ! this.props.subscriptions.length;
+	}
+
+	renderConciergeBanner() {
+		const { nextAppointment, availableSessions, isUserBlocked } = this.props;
+		return (
+			<PurchaseListConciergeBanner
+				nextAppointment={ nextAppointment ?? undefined }
+				availableSessions={ availableSessions }
+				isUserBlocked={ isUserBlocked }
+			/>
+		);
+	}
+
+	renderMembershipSubscriptions() {
+		const { subscriptions } = this.props;
+
+		if ( ! subscriptions.length || this.isDataLoading() ) {
+			return null;
+		}
+
+		return getSubscriptionsBySite( subscriptions ).map( ( site ) => (
+			<MembershipSite site={ site } key={ site.id } />
+		) );
+	}
+
+	render() {
+		const { purchases, sites, translate, subscriptions } = this.props;
+		let content;
+
+		if ( this.isDataLoading() ) {
+			content = <PurchasesSite isPlaceholder />;
+		}
+
+		const onChangeView = () => {
+			alert( 'You clicked something!!' );
+		};
+
+		const getItemId = ( item: Purchases.Purchase ) => {
+			return item.id.toString();
+		};
+
+		if ( purchases && purchases.length ) {
+			if ( useDataViewPurchasesList ) {
```

Comment:
Isn't this always true when rendering this component?

Reply:
That is...true. Since the component is rendered based on `useDataViewPurchasesList` in controller.jsx this will always be true.

I'll remove the secondary conditional, since that is a fallback to rendering the original Purchases List view.